src/Hamiltonians/GutzwillerSampling.jl

```diff
@@ -110,3 +118,58 @@ function Base.getindex(h::GutzwillerOffdiagonals{F,T,A,G}, i)::Tuple{F,T} where
 end
 
 Base.size(h::GutzwillerOffdiagonals) = size(h.offdiagonals)
+
+# the similarity transformed operator `f A f` to calculate expectation value of an operator A
+struct SimTransOperator{T,K<:AbstractHamiltonian{T},O<:Union{AbstractHamiltonian{T},Nothing}} <: AbstractHamiltonian{T}
```

Comment:
```suggestion
struct SimTransOperator{T,K<:AbstractHamiltonian,O<:Union{AbstractHamiltonian,Nothing}} <: AbstractHamiltonian{T}
```
I think the `T` should be determined in the constructor and not restricted in the type. The Hamiltonian and the operator may have different eltypes (for example `G2Correlator` has a complex eltype).